tensorflow/core/grappler/utils/pattern_utils.cc

```diff
@@ -23,17 +23,35 @@ namespace tensorflow {
 namespace grappler {
 namespace utils {
 
-inline const bool IsCommutativeOp(const string& op) {
+const bool IsCommutativeOp(const string& op) {
   // TODO(intel-tf): Add more ops to this list if needed.
+  std::vector<string> op_list = str_util::Split(op, '|');
   static const auto* commutative_ops =
-      new absl::flat_hash_set<string>({"Add", "AddV2", "Mul"});
-  return commutative_ops->contains(op);
+      new absl::flat_hash_set<string>({"Add", "AddV2", "Mul",
```

Comment:
This function should defer to the function here: 

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/grappler/op_types.h#L219

which relies on the canonical op definitions.

Reply:
Perhaps the NodeDef's are available at the call sites to this function, and we can get rid of it entirely and call IsCommutative directly?

Reply:
We would like to restrict the list of commutative ops only to those that have been tested well with the pattern matcher in order to avoid unforeseen issues with fusions, which is why we have a separate function definition of IsCommutativeOp(...).